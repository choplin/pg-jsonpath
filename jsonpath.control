comment = 'JSONPath Wrapper'
default_version = '0.1'
module_pathname = '$libdir/pg-jsonpath'
relocatable = true
