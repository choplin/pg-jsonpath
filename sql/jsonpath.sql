CREATE EXTENSION plv8;
CREATE EXTENSION jsonpath;

SELECT jsonPath('{"x": {"a":1, "b":2}}'::json, '$.x.[a,b]');

SELECT jsonPath('{"x": {"a":1, "b":2}}'::json, '$.x.[a,b]', '{"resultType":"PATH"}'::json);
