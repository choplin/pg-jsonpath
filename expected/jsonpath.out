CREATE EXTENSION plv8;
CREATE EXTENSION jsonpath;
SELECT jsonPath('{"x": {"a":1, "b":2}}'::json, '$.x.[a,b]');
 jsonpath 
----------
 {1,2}
(1 row)

SELECT jsonPath('{"x": {"a":1, "b":2}}'::json, '$.x.[a,b]', '{"resultType":"PATH"}'::json);
         jsonpath          
---------------------------
 {$['x']['a'],$['x']['b']}
(1 row)

